frontend/src/pages/NewPipelineVersion.tsx

```diff
@@ -632,19 +632,34 @@ class NewPipelineVersion extends Page<{}, NewPipelineVersionState> {
     // (1) new pipeline (and a default version) from local file
     // (2) new pipeline (and a default version) from url
     // (3) new pipeline version (under an existing pipeline) from url
-    const { fileName, pipeline, pipelineVersionName, packageUrl, newPipeline } = this.state;
+    const {
+      fileName,
+      pipeline,
+      pipelineVersionName,
+      packageUrl,
+      newPipeline,
+      pipelineName,
+    } = this.state;
     try {
       if (newPipeline) {
         if (!packageUrl && !fileName) {
           throw new Error('Must specify either package url  or file in .yaml, .zip, or .tar.gz');
         }
+        if (!pipelineName) {
+          throw new Error('Pipeline name is required');
+        }
+        this._isValidName(pipelineName!, 'Pipeline name');
       } else {
         if (!pipeline) {
           throw new Error('Pipeline is required');
         }
+        this._isValidName(pipelineName!, 'Pipeline name');
```

Comment:
Question: If this is not a new pipeline: which means `newPipeline` is false, can user still be able to change the pipeline name? Or can we assume that pipeline name is going to be valid because backend has already accepted it?

Reply:
For upload new version (not a new pipeline), the user can only choose pipeline that is existing in the backend, so I think we can remove the pipeline name check here for old pipeline!